build_cmake/Framework/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+		${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/boost-cmake
+		${CMAKE_CURRENT_BINARY_DIR}/boost-cmake
+	)
+endif()
+
+if(ANDROID)
+	if(NOT TARGET cpufeatures)
+		include_directories("${CMAKE_ANDROID_NDK}/sources/android/cpufeatures")
+	endif()
+endif()
+
+if (WIN32)
+	add_definitions(-D_UNICODE -DUNICODE)
+	add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+	add_definitions(-D_SCL_SECURE_NO_WARNINGS)
+	add_definitions(-D_LIB)
+endif (WIN32)
+
+include_directories(../../include)
+
+set(COMMON_SRC_FILES
+	../../src/Base64.cpp
+	../../src/bitmap/Bitmap.cpp
+	../../src/bitmap/BMP.cpp
+	../../src/bitmap/JPEG.cpp
+	../../src/bitmap/PNG.cpp
+	../../src/bitmap/TGA.cpp
+	../../src/BitStream.cpp
+	../../src/Config.cpp
+	../../src/Endian.cpp
+	../../src/idct/IEEE1180.cpp
+	../../src/layout/FlatLayout.cpp
+	../../src/layout/GridLayout.cpp
+	../../src/layout/HorizontalLayout.cpp
+	../../src/layout/LayoutBase.cpp
+	../../src/layout/LayoutBaseItem.cpp
+	../../src/layout/LayoutObject.cpp
+	../../src/layout/LayoutStretch.cpp
+	../../src/layout/VerticalLayout.cpp
+	../../src/LzAri.cpp
+	../../src/math/MathStringUtils.cpp
+	../../src/MemStream.cpp
+	../../src/mpeg2/CodedBlockPatternTable.cpp
+	../../src/mpeg2/DcSizeChrominanceTable.cpp
+	../../src/mpeg2/DcSizeLuminanceTable.cpp
+	../../src/mpeg2/DctCoefficientTable.cpp
+	../../src/mpeg2/DctCoefficientTable0.cpp
+	../../src/mpeg2/DctCoefficientTable1.cpp
+	../../src/mpeg2/InverseScanTable.cpp
+	../../src/mpeg2/MacroblockAddressIncrementTable.cpp
+	../../src/mpeg2/MacroblockTypeBTable.cpp
+	../../src/mpeg2/MacroblockTypeITable.cpp
+	../../src/mpeg2/MacroblockTypePTable.cpp
+	../../src/mpeg2/MotionCodeTable.cpp
+	../../src/mpeg2/QuantiserScaleTable.cpp
+	../../src/mpeg2/VLCTable.cpp
+	../../src/PathUtils.cpp
+	../../src/PtrStream.cpp
+	../../src/Stream.cpp
+	../../src/StreamBitStream.cpp
+	../../src/StdStream.cpp
+	../../src/StdStreamUtils.cpp
+	../../src/Utf8.cpp
+	../../src/xml/FilteringNodeIterator.cpp
+	../../src/xml/Node.cpp
+	../../src/xml/Parser.cpp
+	../../src/xml/Utils.cpp
+	../../src/xml/Writer.cpp
+	../../src/zip/ZipArchiveReader.cpp
+	../../src/zip/ZipArchiveWriter.cpp
+	../../src/zip/ZipDeflateStream.cpp
+	../../src/zip/ZipFile.cpp
+	../../src/zip/ZipInflateStream.cpp
+	../../src/zip/ZipStoreStream.cpp
+)
+if (WIN32)
+	set(PLATFORM_SPECIFIC_SRC_FILES
+		../../src/directinput/Device.cpp 
+		../../src/directinput/Joystick.cpp
+		../../src/directinput/Keyboard.cpp
+		../../src/directinput/Manager.cpp
+		../../src/string_cast.cpp
+	)
+endif (WIN32)
+
+if(ANDROID)
+	set(PLATFORM_SPECIFIC_SRC_FILES
+		../../src/android/AssetStream.cpp
+		../../src/android/AssetManager.cpp
+		../../src/PosixFileStream.cpp
+	)
+endif()
+
+if(NOT ANDROID)
+	set(PLATFORM_SPECIFIC_SRC_FILES
+		${PLATFORM_SPECIFIC_SRC_FILES}
+		
+		../../src/string_cast_sjis.cpp
+	)
+endif()
+
+add_library(Framework ${COMMON_SRC_FILES} ${PLATFORM_SPECIFIC_SRC_FILES})
+set_target_properties(Framework PROPERTIES COMPILE_FLAGS "${CXX_FLAGS}")
```

Comment:
@Thunder07 Is this necessary? Aren't flags set automatically if we change `CMAKE_CXX_FLAGS`?